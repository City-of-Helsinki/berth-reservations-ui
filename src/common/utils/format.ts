export const formatDimension = (value: number | null | undefined, locale: string) => {
  if (!value) return '-';

  const localizedValues = new Intl.NumberFormat(locale, {
    style: 'decimal',
    minimumFractionDigits: 1,
  }).format(value);

  return `${localizedValues} m`;
};

export const formatWeight = (value: number | null | undefined, locale: string) => {
  if (!value) return '-';

  const localizedValues = new Intl.NumberFormat(locale, {
    style: 'decimal',
  }).format(value);

  return `${localizedValues} kg`;
};

export const formatDate = (date: string | null, locale: string, withTime = false) => {
  if (!date) return '-';

  const dateOpts = {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
  };
  const timeOpts = {
    hour: '2-digit',
    minute: '2-digit',
  };
  const options = withTime ? { ...dateOpts, ...timeOpts } : dateOpts;

  return new Date(date).toLocaleString(locale, options);
};

export const formatPrice = (value: number, locale: string, percentage?: number) => {
  const formatter = new Intl.NumberFormat(locale, {
    style: 'currency',
    currency: 'EUR',
    minimumIntegerDigits: 1,
  });

  if (percentage) {
    return `${percentage}%\u00A0\u00A0${formatter.format(value)}`;
  }
  return formatter.format(value);
};

export const formatPercentage = (value: number, locale: string) => {
  return new Intl.NumberFormat(locale, {
    style: 'percent',
  }).format(value / 100);
};
