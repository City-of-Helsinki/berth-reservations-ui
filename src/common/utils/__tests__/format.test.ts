import { formatDate, formatDimension, formatPercentage, formatPrice } from '../format';

describe('format', () => {
  describe('formatDimension', () => {
    it('should add at least one fraction number', () => {
      expect(formatDimension(1, 'fi')).toMatch('1,0');
    });

    it('should add "m" unit to to the supplied value', () => {
      expect(formatDimension(1, 'fi')).toMatch('m');
    });

    it('should return "-" if the value is null', () => {
      expect(formatDimension(null, 'fi')).toBe('-');
      expect(formatDimension(null, 'fi')).toBe('-');
    });
  });

  describe('formatDate', () => {
    it('should format the date according to the supplied locale', () => {
      expect(formatDate('2018-11-28T12:26:28.146227+00:00', 'fi')).toMatchSnapshot();
    });

    it('should include time info when the third argument is true', () => {
      expect(formatDate('2018-11-28T12:26:28.146227+00:00', 'fi', true)).toMatchSnapshot();
    });
  });

  describe('formatPrice', () => {
    it('should add at least two decimals', () => {
      expect(formatPrice(1, 'fi')).toMatch('1,00');
    });

    it('should add the currency unit to to the supplied value', () => {
      expect(formatPrice(1, 'fi')).toMatch('€');
    });

    it('should format prices with percentage correctly', () => {
      expect(formatPrice(1, 'fi', 0.5)).toEqual('0.5%  01,00 €');
    });
  });

  describe('formatPercentage', () => {
    it('should add the percentage sign to to the supplied value', () => {
      expect(formatPercentage(1, 'fi')).toMatch('%');
    });

    it('should return the right value', () => {
      expect(formatPercentage(10, 'fi')).toEqual('10 %');
    });
  });
});
