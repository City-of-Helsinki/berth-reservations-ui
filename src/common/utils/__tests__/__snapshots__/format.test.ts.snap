// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format formatDate should format the date according to the supplied locale 1`] = `"28.11.2018"`;

exports[`format formatDate should include time info when the third argument is true 1`] = `"28.11.2018 klo 12.26"`;
